### ISSUE TITLE
Buttons aren't binding

### ISSUE THREAD

**Issue by Pdubbs610** [sicsix/BYOJoystick#20]
I tried running this and when i went to set the key binds for my thrustmaster T.16000M there was no input being detected. I also couldn't get out of the key bind pop up without alt+f4ing. Is there something i need to do in the files? or is there a video showing the set up process? Edit: the only thing that binds is the keyboard. nothing works on my hotas.

**Comment by sicsix** [comment]
That's odd. If you have Steam Input enabled try turning it off, it's known to cause issues with DirectInput.

Can you post your log files?

**Comment by Pdubbs610** [comment]
i tried to disable steam input and it didn't work. Where are the log files located?

**Comment by Pdubbs610** [comment]
[GameLog_638634067054484436.txt](https://github.com/user-attachments/files/17266655/GameLog_638634067054484436.txt)
[GameLog_638634070176785177.txt](https://github.com/user-attachments/files/17266656/GameLog_638634070176785177.txt)
[GameLog_638634085783000999.txt](https://github.com/user-attachments/files/17266657/GameLog_638634085783000999.txt)
[GameLog_638634088909038405.txt](https://github.com/user-attachments/files/17266658/GameLog_638634088909038405.txt)
[GameLog_638635727769261838.txt](https://github.com/user-attachments/files/17266659/GameLog_638635727769261838.txt)
[GameLog_638636753182183268.txt](https://github.com/user-attachments/files/17266660/GameLog_638636753182183268.txt)

**Comment by sicsix** [comment]
There's a shortcut to Player.log in your VTOL VR installation directory.

**Comment by Pdubbs610** [comment]
[Player.log](https://github.com/user-attachments/files/17271114/Player.log)
is this it?

**Comment by sicsix** [comment]
Yeah that's it, the issue is this:
```
MissingMethodException: void SharpDX.DirectInput.CustomDevice`3.GetBufferedData(System.Collections.Generic.List`1<!2>&)
  at BYOJoystick.UI.BYOJBindingModal.Update () [0x00040] in <78378a88f5274ec594e4b4fc5f50913c>:0 
```

I've got a custom SharpDX.dll included with the mod on Steam that has an extra method added to it. I think your problem might be caused by the wrong version from another mod being loaded first.

Have a look in the workshop folder for VTOL VR at `...\Steam\steamapps\workshop\content\3018410` and see if you can find multiple copies of SharpDX.dll.

**Comment by Pdubbs610** [comment]
there is only 1 'SharpDX.dll' but there is another file named 'SharpDX.DirectInput.dll' do you think that the dot after the SharpDX is causing the issue? Just tried running it only running that mod and it still doesn't work.

**Comment by sicsix** [comment]
Sorry, SharpDX.DirectInput.dll is the one you're looking for. Search in `...\Steam\steamapps\common\VTOL VR` as well, you've got an older version of that DLL somewhere that is getting loaded first by the mod loader. The one in `...\Steam\steamapps\workshop\content\3018410\3276352374` is the one from this mod that you need to keep.

If all else fails, just unsubscribe from all your mods, uninstall VTOL VR, and delete whatever's left in the VTOL VR installation directory and reinstall everything and it should work.

**Comment by Pdubbs610** [comment]
yeah. i got another SharpDX.dll and SharpDX.DirectInput.dll in my VTOLVR_Data\Managed

**Comment by sicsix** [comment]
Just delete those two and you should be good to go, they were likely copied there by the old mod loader from a previous mod you had installed.

**Comment by Pdubbs610** [comment]
that worked. thanks so much

**Comment by Pdubbs610** [comment]
where should the input file be saved to. just in case.

**Comment by sicsix** [comment]
You won't need them.
